File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.c
// Last updated: 10/4/2025, 5:45:12 PM
int numberOfSteps(int num) {
    int count = 0;
    while (num != 0) {
        if (num % 2 == 0) {
            num = num / 2;
        } else {
            num = num - 1;
        }
        count++;
    }
    return count;
}

File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.py
# Last updated: 10/4/2025, 5:45:15 PM
class Solution(object):
    def numberOfSteps(self, num):
        """
        :type num: int
        :rtype: int
        """
        count = 0
        while num != 0:
            if num % 2 == 0:
                num = num // 2   # Use integer division
            else:
                num = num - 1
            count += 1
        return count
